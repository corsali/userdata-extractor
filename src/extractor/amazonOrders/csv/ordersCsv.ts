import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { Orders } from "../models/orders.js";

const extractPrice = (price: string) =>
  parseFloat(/[\d,]*\.[\d]*/.exec(price)?.[0]);

class OrdersCsv extends CsvExtractor {
  async process() {
    await this.parse();

    if (this.csvDocument) {
      const mappedRow = this.csvDocument.map(
        (order: any) =>
          new Orders({
            orderDate: order["order date"],
            orderId: order["order id"],
            title: order.title,
            category: order.category,
            asinIsbn: order["asin/isbn"],
            unspscCode: order["unspsc code"],
            website: order.website,
            releaseDate: order["release date"],
            condition: order.condition,
            seller: order.seller,
            sellerCredentials: order["seller credentials"],
            listPricePerUnit: extractPrice(order["list price per unit"]),
            purchasePricePerUnit: extractPrice(
              order["purchase price per unit"]
            ),
            quantity: order.quantity,
            paymentInstrumentType: order["payment instrument type"],
            purchaseOrderNumber: order["purchase order number"],
            poLineNumber: order["po line number"],
            orderingCustomerEmail: order["ordering customer email"],
            shipmentDate: order["shipment date"],
            shippingAddressName: order["shipping address name"],
            shippingAddressStreet1: order["shipping address street 1"],
            shippingAddressStreet2: order["shipping address street 2"],
            shippingAddressCity: order["shipping address city"],
            shippingAddressState: order["shipping address state"],
            shippingAddressZip: order["shipping address zip"],
            orderStatus: order["order status"],
            carrierNameAndTrackingNumber:
              order["carrier name & tracking number"],
            itemSubtotal: extractPrice(order["item subtotal"]),
            itemSubtotalTax: extractPrice(order["item subtotal tax"]),
            itemTotal: extractPrice(order["item total"]),
            taxExemptionApplied: order["tax exemption applied"],
            taxExemptionType: order["tax exemption type"],
            exemptionOptOut: order["exemption opt-out"],
            buyerName: order["buyer name"],
            currency: order.currency,
            groupName: order["group name"],
          })
      );

      this.table.rows.push(...mappedRow);
    }
  }
}

export const ordersCsv = new OrdersCsv(
  config.SERVICE_AMAZON_ORDERS,
  ".*\\.csv",
  "orders"
);
