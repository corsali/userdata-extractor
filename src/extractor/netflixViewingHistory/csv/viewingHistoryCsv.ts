import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { ViewingHistory } from "../models/viewingHistory.js";

class ViewingHistoryCsv extends CsvExtractor {
  async process() {
    await this.parse();

    const mappedRow = this.csvDocument.map(
      (row: any) => new ViewingHistory(row.title, row.date)
    );

    this.table.rows.push(...mappedRow);
  }
}

export const viewingHistoryCsv = new ViewingHistoryCsv(
  config.SERVICE_NETFLIX_VIEWING_HISTORY,
  ".*NetflixViewingHistory.csv",
  "viewing_history"
);
