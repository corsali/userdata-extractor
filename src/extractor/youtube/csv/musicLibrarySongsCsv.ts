import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { MusicLibrarySongs } from "../models/musicLibrarySongs.js";

class MusicLibrarySongsCsv extends CsvExtractor {
  async process() {
    this.parse();

    const mappedRow = this.csvDocument.map(
      (row: any) =>
        new MusicLibrarySongs({
          songUrl: row["song url"],
          songTitle: row["song title"],
          albumTitle: row["album title"],
          artistNames: row["artist names"],
        })
    );

    this.table.rows.push(...mappedRow);
  }
}

export const musicLibrarySongsCsv = new MusicLibrarySongsCsv(
  config.SERVICE_YOUTUBE,
  ".*/music-library-songs.csv",
  "music_library_songs"
);
