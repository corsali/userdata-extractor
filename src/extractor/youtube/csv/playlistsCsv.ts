import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { Playlists } from "../models/playlists.js";

class PlaylistsCsv extends CsvExtractor {
  async process() {
    const playlistHeader = await this.parseText(
      this.fileContents.split("\n").slice(0, 2).join("\n")
    );

    const playlistItems = await this.parseText(
      this.fileContents.split("\n").slice(2).join("\n")
    );

    const mappedRow = playlistItems.map(
      (playlistItem: any) =>
        new Playlists({
          playlistId: playlistHeader[0]["playlist id"],
          channelId: playlistHeader[0]["channel id"],
          dateCreated: playlistHeader[0]["time created"],
          dateUpdated: playlistHeader[0]["time updated"],
          playlistTitle: playlistHeader[0].title,
          playlistDescription: playlistHeader[0].description,
          playlistVisibility: playlistHeader[0].visibility,
          itemVideoId: playlistItem["video id"],
          dateItemVideoAdded: playlistItem["time added"],
        })
    );

    this.table.rows.push(...mappedRow);
  }
}

export const playlistsCsv = new PlaylistsCsv(
  config.SERVICE_YOUTUBE,
  "playlists/[^/]*\\.csv",
  "playlists"
);
