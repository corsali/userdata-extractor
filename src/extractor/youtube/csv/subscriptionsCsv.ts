import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { Subscriptions } from "../models/subscriptions.js";

class SubscriptionsCsv extends CsvExtractor {
  async process() {
    this.parse({ skipEmptyLines: true });

    const mappedRow = this.csvDocument.map(
      (row: any) =>
        new Subscriptions(
          row["Channel Id"],
          row["Channel Url"],
          row["Channel Title"]
        )
    );

    this.table.rows.push(...mappedRow);
  }
}

export const subscriptionsCsv = new SubscriptionsCsv(
  config.SERVICE_YOUTUBE,
  ".*/subscriptions.csv",
  "subscriptions"
);
