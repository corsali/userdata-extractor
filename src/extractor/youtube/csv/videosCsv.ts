import config from "../../../config/index.js";
import { CsvExtractor } from "../../csvExtractor.js";
import { Videos } from "../models/videos.js";

class VideosCsv extends CsvExtractor {
  async process() {
    await this.parse();

    const mappedRow = this.csvDocument.map(
      (row: any) =>
        new Videos({
          videoId: row["video id"],
          channelId: row["channel id"],
          title: row.title,
          status: row.status,
          visibility: row.visibility,
          timeCreated: row["time created"],
          timePublished: row["time published"],
          duration: row.duration,
          description: row.description,
          category: row.category,
          locationAddress: row["location address"],
          locationCountry: row["location country"],
          timeRecorded: row["time recorded"],
          likeCount: row["like count"],
          dislikeCount: row["dislike count"],
          viewCount: row["view count"],
          syndication: row.syndication,
          license: row.license,
          embeddable: row.embeddable,
        })
    );

    this.table.rows.push(...mappedRow);
  }
}

export const videosCsv = new VideosCsv(
  config.SERVICE_YOUTUBE,
  ".*/video metadata.csv",
  "videos"
);
