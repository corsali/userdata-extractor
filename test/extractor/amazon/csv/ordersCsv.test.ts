import { ordersCsv } from "../../../../src/extractor/amazon/csv/ordersCsv";
import { loadTestFileAsText } from "../../../helper";

describe("Orders (CSV)", () => {
  test("it should load file correctly", async () => {
    const data = await loadTestFileAsText("/csv/amazon/Amazon_Orders_csv.csv");

    ordersCsv.fileContents = data;

    ordersCsv.process();
    const { rows } = ordersCsv.table;

    expect(rows).toMatchSnapshot();
  });
});
