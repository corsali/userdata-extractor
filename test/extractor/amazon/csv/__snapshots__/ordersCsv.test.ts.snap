// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders (CSV) it should load file correctly 1`] = `
Array [
  Orders {
    "asin_isbn": TextTableValue {
      "type": "text",
      "value": "B09JR24YYW",
    },
    "buyer_name": TextTableValue {
      "type": "text",
      "value": "Z Hay",
    },
    "carrier_name_and_tracking_number": TextTableValue {
      "type": "text",
      "value": "AMZN_US(TBA154327887604)",
    },
    "category": TextTableValue {
      "type": "text",
      "value": "CHARGING_ADAPTER",
    },
    "condition": TextTableValue {
      "type": "text",
      "value": "new",
    },
    "currency": TextTableValue {
      "type": "text",
      "value": "USD",
    },
    "exemption_opt_out": BoolTableValue {
      "type": "bool",
      "value": false,
    },
    "group_name": TextTableValue {
      "type": "text",
      "value": "",
    },
    "item_subtotal": FloatTableValue {
      "type": "float",
      "value": 59,
    },
    "item_subtotal_tax": FloatTableValue {
      "type": "float",
      "value": 5.22,
    },
    "item_total": FloatTableValue {
      "type": "float",
      "value": 64.22,
    },
    "list_price_per_unit": FloatTableValue {
      "type": "float",
      "value": 59,
    },
    "order_date": DateTableValue {
      "type": "date",
      "value": 1641859200000,
    },
    "order_id": TextTableValue {
      "type": "text",
      "value": "113-9614127-1313059",
    },
    "order_status": TextTableValue {
      "type": "text",
      "value": "Shipped",
    },
    "ordering_customer_email": EmailTableValue {
      "type": "email",
      "value": "zach@gmail.com",
    },
    "payment_instrument_type": TextTableValue {
      "type": "text",
      "value": "Visa - 5843",
    },
    "po_line_number": TextTableValue {
      "type": "text",
      "value": "",
    },
    "purchase_order_number": TextTableValue {
      "type": "text",
      "value": "",
    },
    "purchase_price_per_unit": FloatTableValue {
      "type": "float",
      "value": 59,
    },
    "quantity": IntegerTableValue {
      "type": "integer",
      "value": 1,
    },
    "release_date": DateTableValue {
      "type": "date",
      "value": null,
    },
    "seller": TextTableValue {
      "type": "text",
      "value": "Amazon.com",
    },
    "seller_credentials": TextTableValue {
      "type": "text",
      "value": "",
    },
    "shipment_date": DateTableValue {
      "type": "date",
      "value": 1642032000000,
    },
    "shipping_address_city": TextTableValue {
      "type": "text",
      "value": "BLOOMINGTON",
    },
    "shipping_address_name": TextTableValue {
      "type": "text",
      "value": "Z Hay",
    },
    "shipping_address_state": TextTableValue {
      "type": "text",
      "value": "CO",
    },
    "shipping_address_street_1": TextTableValue {
      "type": "text",
      "value": "123 Street APT B",
    },
    "shipping_address_street_2": TextTableValue {
      "type": "text",
      "value": "",
    },
    "shipping_address_zip": TextTableValue {
      "type": "text",
      "value": "80305-6267",
    },
    "tax_exemption_applied": BoolTableValue {
      "type": "bool",
      "value": false,
    },
    "tax_exemption_type": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "Apple 67W USB-C Power Adapter",
    },
    "unspsc_code": TextTableValue {
      "type": "text",
      "value": "26111704",
    },
    "website": UrlTableValue {
      "type": "url",
      "value": "Amazon.com",
    },
  },
  Orders {
    "asin_isbn": TextTableValue {
      "type": "text",
      "value": "B01MQ5Z080",
    },
    "buyer_name": TextTableValue {
      "type": "text",
      "value": "Z Hay",
    },
    "carrier_name_and_tracking_number": TextTableValue {
      "type": "text",
      "value": "AMZN_US(TBA154327887604)",
    },
    "category": TextTableValue {
      "type": "text",
      "value": "ELECTRONIC_CABLE",
    },
    "condition": TextTableValue {
      "type": "text",
      "value": "new",
    },
    "currency": TextTableValue {
      "type": "text",
      "value": "USD",
    },
    "exemption_opt_out": BoolTableValue {
      "type": "bool",
      "value": false,
    },
    "group_name": TextTableValue {
      "type": "text",
      "value": "",
    },
    "item_subtotal": FloatTableValue {
      "type": "float",
      "value": 18.48,
    },
    "item_subtotal_tax": FloatTableValue {
      "type": "float",
      "value": 1.63,
    },
    "item_total": FloatTableValue {
      "type": "float",
      "value": 20.11,
    },
    "list_price_per_unit": FloatTableValue {
      "type": "float",
      "value": 19,
    },
    "order_date": DateTableValue {
      "type": "date",
      "value": 1641859200000,
    },
    "order_id": TextTableValue {
      "type": "text",
      "value": "113-9614127-1313059",
    },
    "order_status": TextTableValue {
      "type": "text",
      "value": "Shipped",
    },
    "ordering_customer_email": EmailTableValue {
      "type": "email",
      "value": "zach@gmail.com",
    },
    "payment_instrument_type": TextTableValue {
      "type": "text",
      "value": "Visa - 5843",
    },
    "po_line_number": TextTableValue {
      "type": "text",
      "value": "",
    },
    "purchase_order_number": TextTableValue {
      "type": "text",
      "value": "",
    },
    "purchase_price_per_unit": FloatTableValue {
      "type": "float",
      "value": 18.48,
    },
    "quantity": IntegerTableValue {
      "type": "integer",
      "value": 1,
    },
    "release_date": DateTableValue {
      "type": "date",
      "value": null,
    },
    "seller": TextTableValue {
      "type": "text",
      "value": "Amazon.com",
    },
    "seller_credentials": TextTableValue {
      "type": "text",
      "value": "",
    },
    "shipment_date": DateTableValue {
      "type": "date",
      "value": 1642032000000,
    },
    "shipping_address_city": TextTableValue {
      "type": "text",
      "value": "BLOOMINGTON",
    },
    "shipping_address_name": TextTableValue {
      "type": "text",
      "value": "Z Hay",
    },
    "shipping_address_state": TextTableValue {
      "type": "text",
      "value": "CO",
    },
    "shipping_address_street_1": TextTableValue {
      "type": "text",
      "value": "123 Street DR APT B",
    },
    "shipping_address_street_2": TextTableValue {
      "type": "text",
      "value": "",
    },
    "shipping_address_zip": TextTableValue {
      "type": "text",
      "value": "80305-6267",
    },
    "tax_exemption_applied": BoolTableValue {
      "type": "bool",
      "value": false,
    },
    "tax_exemption_type": TextTableValue {
      "type": "text",
      "value": "",
    },
    "title": TextTableValue {
      "type": "text",
      "value": "Apple USB-C Charge Cable (2m)",
    },
    "unspsc_code": TextTableValue {
      "type": "text",
      "value": "26111700",
    },
    "website": UrlTableValue {
      "type": "url",
      "value": "Amazon.com",
    },
  },
]
`;
